### ISSUE TITLE
Add missing data table header markup for device code lookup page

### ISSUE THREAD

**Issue by emyl3** [CDCgov/prime-simplereport#8571]
## Area or Component
`/app/results/upload/submit/code-lookup` aka Device code lookup page

## Quick Summary
 The `test_result` table and `specimen_type` tables are missing header cell markup.

### How to Replicate

- Go to the device code lookup page.
- Search for and find a device that will return both the `test_result` and `specimen_type` tables.
- Notice that both tables do not have `<th>` markup.

## Accessibility Violation

[semantic-data-table-headers](https://docs.deque.com/issue-help/1.0.0/en/semantic-data-table-headers)

## Action Requested

Data tables must use semantic markup to associate data cells with their corresponding header cell(s).

## Screenshots (optional)

<img width="1635" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMzBZQ1E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--226e6b1abfeb9044a6ce1c318cdcc4c8e7a927eb/Screenshot%202025-03-03%20at%2015.05.38.png" alt="Screenshot 2025-03-03 at 15.05.38.png" />

## Source Code (optional)

[which file or line is the issue attached to?]: #
